Source/Core/Core/IOS/USB/Common.h

```diff
@@ -109,6 +134,49 @@ struct IntrMessage : TransferCommand
   u8 endpoint = 0;
   using TransferCommand::TransferCommand;
 };
+
+struct IsoMessage : TransferCommand
+{
+  u32 packet_sizes_addr = 0;
+  std::vector<u16> packet_sizes;
+  u16 length = 0;
+  u8 num_packets = 0;
+  u8 endpoint = 0;
+  using TransferCommand::TransferCommand;
+  void SetPacketReturnValue(size_t packet_num, u16 return_value) const;
+};
+
+class Device
+{
+public:
+  virtual ~Device() = default;
```

Comment:
This should go in the cpp file, otherwise the vtable will be generated in every translation unit that uses the class. Placing:

```cpp
Device::Device() = default;
```

in the cpp file and making the class definition

```cpp
virtual ~Device();
```

is sufficient.